### PR TITLE
[CodingStyle]  Escape content delimiter in pattern for ConsistentPregDelimiterRector

### LINKED ISSUE #5987: Incorrect behavior of ConsistentPregDelimiterRector
# Bug Report

| Subject        | Details              |
| :------------- | :--------------------|
| Rector version | last dev-master      |
| Installed as   | composer dependency  |

## Minimal PHP Code Causing Issue

See https://getrector.org/demo/35bbca53-12af-4f23-9621-1c3b8e07ee0f

```php
<?php

preg_match('/^#(([a-f0-9]{3}){1,2})$/i', '#ffffff');
```

### Responsible rules

* `ConsistentPregDelimiterRector`

## Expected Behavior

The regex includes the new delimeter. This makes the regex invalid after running the ConsistentPregDelimiterRector.
The ConsistentPregDelimiterRector should escape the new delimeter inside the regex.

### PR DESCRIPTION
**Pull request by samsonasik**
Fixes #5987

### CODE REVIEW

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

File: rules/CodingStyle/Rector/FuncCall/ConsistentPregDelimiterRector.php [hunk 1]
```diff
@@ -168,6 +168,12 @@ private function refactorArgument(Arg $arg): void
 
         $string->value = Strings::replace($value, self::INNER_REGEX, function (array $match): string {
             $innerPattern = $match['content'];
+            $positionDelimiter = strpos($innerPattern, $this->delimiter);
+
+            if ($positionDelimiter > 0) {
```
Comment (staabm) [inline rules/CodingStyle/Rector/FuncCall/ConsistentPregDelimiterRector.php:173]:
```suggestion
            if ($positionDelimiter !== false) {
```

?

**Comment by TomasVotruba** [comment]
Thanks :+1:

**Pull request merged**